crates/node/src/interactive_cli/context.rs

```diff
@@ -165,22 +179,57 @@ impl ContextCommand {
             Commands::Join {
                 private_key,
                 invitation_payload,
+                context,
+                identity,
             } => {
+                let public_key = private_key.public_key();
+
                 let response = node
                     .ctx_manager
                     .join_context(private_key, invitation_payload)
                     .await?;
 
-                if let Some((context_id, identity)) = response {
+                if let Some((context_id, identity_val)) = response {
+                    // Create identity alias in the context if --name is provided
+                    if let (Some(name_alias), Some(identity)) =
+                        (context.as_ref(), identity.as_ref())
```

Comment:
we shouldn't need the context to be aliased for the identity to be aliased

```suggestion
                    // Create identity alias in the context if --name is provided
                    if let Some(identity) = identity.as_ref()
```